internal/engine/upper.go

```diff
@@ -451,6 +450,14 @@ func handleBuildCompleted(ctx context.Context, engineState *store.EngineState, c
 }
 
 func appendToTriggerQueue(state *store.EngineState, mn model.ManifestName, reason model.BuildReason) {
+	if mn == model.TiltfileManifestName {
+		// the Tiltfile doesn't get queued like other manifests; instead we mark it as
+		// "pending" in the engine state and it gets picked up by the configs controller
+		state.PendingTiltfileTrigger = time.Now()
```

Comment:
also, as a general rule, we shouldn't use time.Now() in reducers. Actions get queued up, and there's nothing really guaranteeing that time.Now() inside the reducer corresponds well to when the trigger happened. 

Here's a doc explaining this - https://redux.js.org/basics/reducers#handling-actions (the distinction comes up in any reactive framework that needs functions to be pure)